### PR TITLE
Feature/install command plugin update

### LINKED ISSUE #112: Installing plugins
Not sure what others think, but for me it is hard to imagine a world where I wouldn't want my project's CLI plugin as a global alias. There is already great documentation in the plugin example of configuring the global alias for your project. It even mentions the future possibility of installing plugin aliases.

There are various ways this could play out, but imho it would be great to be bundled in as part of the larger 'install' command of the CLI. When that 'install' command is run, it additionally will check your current directory to see if its a project (or it can find one of its parent directories is one) and suggest (or automatically setup) the plugin alias globally `npm install -g .`. 

If for whatever reason, rolling it into the main `install` command would prove difficult, we could simply have a separate `install-plugin` command.

### PR DESCRIPTION
**Pull request by brandongregoryscott**
Fixes #112 Installing plugins

Adds logic to the `install` command to automatically install the CLI globally if the `bin` section of the `package.json` has a key (always taking the first, if multiple found) that does not match `and-cli`. Some minor refactors to pull out each step of that command into a private function for readability & easier extraction to a module later on. 

I wanted to write some integration tests for this, but I ran into some issues managing temporary/home directories in the test suite. I'll do some further investigation and possibly follow up with later PR(s).

-   [x] Related GitHub issue(s) linked in PR description
-   [x] Destination branch merged, built and tested with your changes
-   [x] Code formatted and follows best practices and patterns
-   [x] Code builds cleanly (no _additional_ warnings or errors)
-   [x] Manually tested
-   [x] Automated tests are passing
-   [x] No _decreases_ in automated test coverage
    - More test coverage can certainly be added once the `install` functions are broken out into a module
-   [x] Documentation updated (readme, docs, comments, etc.)
-   [x] Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: and-cli-install.js [hunk 1]
```diff
@@ -85,7 +62,105 @@ require("./command-runner").run(async () => {
...
+        echo.success(`Successfully installed ${CLI_NAME}-dev alias`);
+    };
+
+    const _installLocalProjectGlobally = (binName) => {
+        const cmd = install.cmd(".");
+        echo.message(
+            `Installing current project as a global npm tool... (via ${cmd})`
+        );
+
+        const { code } = shell.exec(cmd);
```
Comment (HeyKos) [inline and-cli-install.js:126]: The code here and on line 88 are identical. This exec and error evaluation could potentially get combined into a shared function.

File: and-cli-install.js [hunk 2]
```diff
@@ -97,7 +172,7 @@ require("./command-runner").run(async () => {
         .parse(process.argv);
 
     // If no options are passed in, performs installation steps
-    if (process.argv.slice(2).length === 0) {
+    if (js.hasNoArguments()) {
```
Comment (HeyKos) [inline and-cli-install.js:175]: Nice little refactor here. Much easier to read!

File: modules/package-config.test.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ jest.mock("find-package-json");
 // #region Imports
 // -----------------------------------------------------------------------------------------
 
-const { CLI_NAME } = require("./constants");
+const { BIN, CLI_NAME, PACKAGE_JSON, ENTRYPOINT } = require("./constants");
```
Comment (HeyKos) [inline modules/package-config.test.js:13]: Excellent job refactoring to bring these in from constants. Overall excellent job adding these tests!

**Review by HeyKos — APPROVED** [review]
Awesome job @brandongregoryscott ! Really nice refactoring done along the way too.

File: and-cli-install.js [hunk 4]
```diff
@@ -85,7 +62,105 @@ require("./command-runner").run(async () => {
...
+
+        echo.success(`Successfully installed ${CLI_NAME} globally`);
+        echo.newLine();
+    };
+
+    const _installDevAlias = () => {
+        echo.message(
+            `Configuring cli development '${CLI_NAME}-dev' bash alias...`
+        );
+        const pathToCli = upath.join(shell.pwd().toString(), ENTRYPOINT);
```
[the remark this answers is not in the document]
Comment (brandongregoryscott) [reply and-cli-install.js:102]: I would certainly hope not 😂 The TS definition files for `shelljs` have the return defined as `string & ShellReturnValue`, and I've never seen it return `null` or `undefined` - I think we're safe 👍

[the remark this answers is not in the document]
Comment (HeyKos) [reply and-cli-install.js:102]: Lol! Goes to show you how much I know about ShellJS. Thanks for humoring my question.

**Pull request merged**